gradle-consistent-versions-idea-plugin/src/test/java/com/palantir/gradle/versions/intellij/VersionPropsCodeInsightTest.java

```diff
@@ -52,31 +53,57 @@ public void test_version_completion() {
     }
 
     @Test
-    public void test_group_completion() throws Exception {
+    public void test_wild_card_version_completion() throws InterruptedException {
+        JavaCodeInsightTestFixture fixture = getFixture();
+        // The file name is required for context but does not need to exist on the filesystem
+        fixture.configureByText("versions.props", "com.palantir.tokens:* = <caret>");
+        fixture.complete(CompletionType.BASIC);
+        Thread.sleep(500);
```

Comment:
Thread.sleep in tests like this is often risky and can make flaky tests. Generally the way to do this is to use [awaitility](https://github.com/awaitility/awaitility) to wait up to say 1 min but try every X milliseconds until it works. It keeps tests running quickly but also means if something takes a unexpectedly long time time (maybe a network request takes 5 seconds?) the test does not fail and so is (much less) flaky

Reply:
good to know thanks!